src/cheri_insts.sail

```diff
@@ -666,19 +672,50 @@ function clause execute (CSeal(cd, cs, ct)) =
 union clause ast = CCSeal : (regidx, regidx, regidx)
 function clause execute (CCSeal(cd, cs, ct)) =
 {
-  let cs_val = readCapReg(cs);
-  let ct_val = readCapReg(ct);
+  let cs_val = readCapReg(cs); /* Tagged but unsealed cap */
+  let ct_val = readCapReg(ct); /* Authority */
   let ct_cursor = getCapCursor(ct_val);
   let (ct_base, ct_top) = getCapBounds(ct_val);
+
   if not (cs_val.tag) then {
     handle_cheri_reg_exception(CapEx_TagViolation, cs);
     RETIRE_FAIL
-  } else if not (ct_val.tag) | ct_val.address == to_bits(sizeof(xlen), otype_unsealed) then {
-    writeCapReg(cd, cs_val);
-    RETIRE_SUCCESS
   } else if cs_val.sealed then {
+    /*
+     * If the input to be conditionally sealed is already sealed, just pass it
+     * through before any other checks are made.
+     */
     handle_cheri_reg_exception(CapEx_SealViolation, cs);
```

Comment:
Comment disagrees with the code here? It seems the comment is correct?

Reply:
`CCSeal` has really not been my finest work, has it?  Sorry!

Reply:
No worries, my first implementations of features were no better before testing!